### PR TITLE
Verify assemblies are built in READYTORUN mode

### PR DESCRIPTION
**Pull request by omajid**
Verify that assemblies are compiled in READYTORUN mode to catch issues like these: https://github.com/dotnet/source-build/issues/1452

This change merges the test previously written in python ("dlls-in-release-mode") with this new test to avoid duplication.

We are mostly interested in assemblies in the shared framework, not in the SDK or targeting packs.

Fixes: #103

cc @RheaAyase @tmds

### CODE REVIEW

**Comment by RheaAyase** [comment]
Oh I missed this sorry, seems good to me. @tmds anything to add? :)

File: assemblies-valid/AssembliesValid.cs [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                p.WaitForExit();
+                using (StreamReader r = p.StandardOutput)
+                {
+                    standardOutput = r.ReadToEnd();
+                }
+                return p.ExitCode;
+            }
+        }
+
+        static Machine GetCurrentMachine()
```
Comment (tmds) [inline assemblies-valid/AssembliesValid.cs]: At some point we may want to put these functions in a shared source folder, so multiple tests can use them.

File: assemblies-valid/AssembliesValid.cs [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        };
+
+        [Fact]
+        public void ValidateAssemblies()
+        {
+            string dotnetPath = null;
+            int exitCode = RunProcessAndGetOutput(new string[] { "command", "-v", "dotnet" }, out dotnetPath);
+            if (exitCode != 0)
+            {
+                Console.Error.WriteLine("'dotnet' command not found");
```
Comment (tmds) [inline assemblies-valid/AssembliesValid.cs]: This test is a Console application in disguise :smile: 
General practice is not to do use `Console` in tests, and split them into different `[Fact]`/`[Theory]`. The `[Fact]` that failed, and the exception thrown should be enough to figure out what is failing.

I'm fine if you want to keep the test as is.

**Comment by tmds** [comment]
Did you verify the test is failing as expected on an non-r2r build (ours), and passing on an r2r build (like Microsoft's)?

**Comment by omajid** [comment]
> Did you verify the test is failing as expected on an non-r2r build (ours), and passing on an r2r build (like Microsoft's)?

Yes. For source-build, it flags all of 2.1.x as okay. It flags the 3.1.x `Microsoft.NETCore.App` as okay, and flags `Microsoft.AspNetCore.App` as being buggy. It flags all of  Microsoft's 3.1.x as okay.

**Comment by omajid** [comment]
@tmds Any other concerns? Is this okay to merge?

**Comment by tmds** [comment]
@omajid this is good to merge!

**Pull request merged**